virtualizarr/codecs.py

```diff
@@ -12,29 +10,39 @@
 
     from .manifests.array import ManifestArray
 
+CodecPipeline = Tuple[
+    Union["ArrayArrayCodec", "ArrayBytesCodec", "BytesBytesCodec"], ...
+]
 
-def get_codecs(
-    array: Union["ManifestArray", "Array"],
-    normalize_to_zarr_v3: bool = False,
-) -> Union[Codec, tuple["ArrayArrayCodec | ArrayBytesCodec | BytesBytesCodec", ...]]:
-    """
-    Get the codecs for either a ManifestArray or a Zarr Array.
-
-    Parameters:
-        array (Union[ManifestArray, ZarrArray]): The input array, either ManifestArray or Zarr Array.
 
-    Returns:
-        List[Optional[Codec]]: A list of codecs or an empty list if no codecs are found.
-
-    Raises:
-        ImportError: If `zarr` is required but not installed.
-        ValueError: If the array type is unsupported.
+def get_codecs(array: Union["ManifestArray", "Array"]) -> CodecPipeline:
```

Comment:
nit to make types clearer, see above comment on zarr being a required dependency

```suggestion
import zarr

def get_codecs(array: Union["ManifestArray", "zarr.Array"]) -> CodecPipeline:
```